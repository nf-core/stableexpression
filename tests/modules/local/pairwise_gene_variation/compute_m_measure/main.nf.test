nextflow_process {

    name "Test Process COMPUTE_M_MEASURE"
    script "modules/local/pairwise_gene_variation/compute_m_measure/main.nf"
    process "COMPUTE_M_MEASURE"
    tag "m_measure"
    tag "module"

    test("Four initial chunk files") {

        when {
            process {
                """
                ch_count_file = Channel.fromPath( '$projectDir/tests/test_data/pairwise_gene_variation/make_chunks/input/counts.parquet', checkIfExists: true)
                ch_ratio_std_files = Channel.fromPath( '$projectDir/tests/test_data/pairwise_gene_variation/compute_m_measure/input/std.*.parquet', checkIfExists: true).collect()
                input[0] = ch_count_file
                input[1] = ch_ratio_std_files
                """
            }
        }

        then {
            assert process.success
            // assert snapshot(process.out).match()
        }

    }

}
