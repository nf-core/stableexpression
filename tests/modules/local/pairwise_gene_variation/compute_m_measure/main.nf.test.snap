{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "m_measures.csv:md5,996ba9ca255d5d4032e30e04bc062079"
                ],
                "1": [
                    [
                        "COMPUTE_M_MEASURE",
                        "python",
                        "3.12.8"
                    ]
                ],
                "2": [
                    [
                        "COMPUTE_M_MEASURE",
                        "polars",
                        "1.17.1"
                    ]
                ],
                "m_measures": [
                    "m_measures.csv:md5,996ba9ca255d5d4032e30e04bc062079"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-16T11:48:03.528113087"
    },
    "Four initial chunk files": {
        "content": [
            {
                "0": [
                    "m_measures.csv:md5,8d6b01ed9f1dd7767f9ac370cec263ad"
                ],
                "1": [
                    [
                        "COMPUTE_M_MEASURE",
                        "python",
                        "3.12.8"
                    ]
                ],
                "2": [
                    [
                        "COMPUTE_M_MEASURE",
                        "polars",
                        "1.17.1"
                    ]
                ],
                "m_measures": [
                    "m_measures.csv:md5,8d6b01ed9f1dd7767f9ac370cec263ad"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-30T16:16:59.347457078"
    }
}